src/server.rs

```diff
@@ -48,14 +43,19 @@ pub fn run_with_config(config: Config) {
     }
 }
 
-async fn run_tokio_runtime(api: ApiState) {
-    let port = api.config.api_port;
-    let app = routes(api);
-    let listener = tokio::net::TcpListener::bind(SocketAddr::from(([0, 0, 0, 0], port)))
+pub fn start_api(state: ApiState, should_close_receiver: Receiver<()>) -> tokio::io::Result<()> {
+    let rt = Runtime::new().unwrap();
+    rt.block_on(async { run_tokio_runtime(state, should_close_receiver).await })
```

Comment:
```suggestion
    rt.block_on(run_tokio_runtime(state, should_close_receiver))
```
isn't this the same?